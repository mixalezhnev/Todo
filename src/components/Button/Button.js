import React, { Component } from 'react';
import '../../style/Button.css';

class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    alert('Oh yea ;)');
  }

  render(props) {
    return (
      <a href="#" className="Button" onClick={this.handleClick}> 
        { this.props.text }
      </a>
    )
  }
}

export default Button;