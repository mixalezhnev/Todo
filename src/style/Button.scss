.Button {
  width: 100px;
  min-height: 36px;
  border: 0;
  outline: none;
  text-align: center;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 0 11px;
  background-color: #1B89DE;
  border-radius: 4px;
  font-family: sans-serif;
  font-weight: 600;
  text-decoration: none;
  color: white;
  margin-bottom: 5px;
}

.Button-red {
  background: #FB8888;
}

.Button-grey {
  background: #5E6472;
}