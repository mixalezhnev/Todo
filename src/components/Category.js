import React from 'react';
import { Link } from 'react-router';

import Button from './Button';
import './Category.res/style.css';

Category.propTypes = {
  id: React.PropTypes.node.isRequired,
  title: React.PropTypes.string.isRequired,
  onExtend: React.PropTypes.oneOfType(
    [
      React.PropTypes.func,
      React.PropTypes.bool
    ]).isRequired,
  children: React.PropTypes.oneOfType(
    [
      React.PropTypes.arrayOf(React.PropTypes.element.isRequired),
      React.PropTypes.bool,
      React.PropTypes.object
    ])
}

export default function Category({ id, title, children, onExtend }) {
  return (
    <div className="my-category-component">
      {onExtend &&
        <Button type="extend" onClick={onExtend} style={{ marginRight: '10px' }}/>}
      <Link to={`/main/${id}/${title}/`}>{title}</Link>
      {children}
    </div>
  );
}
