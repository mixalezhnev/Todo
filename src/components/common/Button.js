import React from 'react';

import './Button.res/style.css'

const Button = (props) => {
  const {type, styles} = props;
  
  return  <button style={styles} className={`Button Button-${type}`}></button>;
}

Button.defaultProps = {
  type: 'add',
  styles: { 
    top: 0,
    left: 0
  }
}

export default Button;
