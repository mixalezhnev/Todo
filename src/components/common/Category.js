import React, { Component } from 'react';
import { Link } from 'react-router';

import Button from '../common/Button';
import './Category.res/style.css';

export default class Category extends Component {
  state = { extended: true }

  static propTypes = {
    title: React.PropTypes.string,
    mayExtends: React.PropTypes.bool.isRequired,
    changeGlobalStorage: React.PropTypes.func.isRequired,
    globalStorage: React.PropTypes.object.isRequired,
    handleExtendClick: React.PropTypes.func.isRequired
  }

  handleEditCategoryClick = () => {
    const {
      title,
      globalStorage,
      changeGlobalStorage
    } = this.props;

    globalStorage.editCategory(title);
    changeGlobalStorage({ globalStorage });
  }

  handleAddNestedCategoryClick = () => {
    const {
      title,
      globalStorage,
      changeGlobalStorage
    } = this.props;

    globalStorage.addNestedCategory(title);
    changeGlobalStorage({ globalStorage });
  }

  handleDeleteCategoryClick = () => {
    const {
      title,
      globalStorage,
      changeGlobalStorage
    } = this.props;

    globalStorage.deleteCategory(title);
    changeGlobalStorage({ globalStorage });
  }

  render() {
    const {
      title,
      mayExtends,
      handleExtendClick
    } = this.props;

    return (
      <div className="my-category-component">
        {mayExtends &&
          <Button type="extend" onClick={handleExtendClick}/>}
        <Link to={`/${title}/`}>{title}</Link>
        <Button
          type="edit"
          categoryName={title}
          onClick={this.handleEditCategoryClick}/>
        <Button
          type="delete"
          onClick={this.handleDeleteCategoryClick}
          categoryName={title}/>
        <Button
          onClick={this.handleAddNestedCategoryClick}
          type="add"
          categoryName={title}/>
      </div>
    );
  }
}
