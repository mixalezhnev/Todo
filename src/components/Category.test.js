import React from 'react';
import { shallow } from 'enzyme';

import { Link } from 'react-router';
import Category from './Category';
import Button from './Button';

const setup = () => {
  const props = {
    id: 0,
    title: 'Films',
    onExtend: () => {}
  };

  const enzymeWrapper = shallow(
    <Category {...props}>
      <div className='test-class'/>
    </Category>
  );

  return {
    props,
    enzymeWrapper
  }
};

describe('components', () => {
  describe('Category', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});