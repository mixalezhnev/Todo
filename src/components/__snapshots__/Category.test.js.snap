exports[`components Category should render self and subcomponents 1`] = `
<div
  className="my-category-component">
  <Button
    onClick={[Function]}
    style={
      Object {
        "marginRight": "10px",
      }
    }
    type="extend" />
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/main/0/Films/">
    Films
  </Link>
  <div
    className="test-class" />
</div>
`;
